src/lib/AirspeedValidator/AirspeedValidator.cpp

```diff
@@ -0,0 +1,277 @@
+		float lpos_vz, float lpos_evh, float lpos_evv, float att_q[4])
+{
+	bool att_valid = true; // att_valid could also be a input_data state
+
+	_wind_estimator.update(time_now_usec);
+
+	if (lpos_valid && att_valid && _in_fixed_wing_flight) {
+
+		Vector3f vI(lpos_vx, lpos_vy, lpos_vz);
+		Quatf q(att_q);
+
+		// sideslip fusion
+		_wind_estimator.fuse_beta(time_now_usec, vI, q);
+
+		// airspeed fusion (with raw TAS)
+		const Vector3f vel_var{Dcmf(q) *Vector3f{lpos_evh, lpos_evh, lpos_evv}};
+		_wind_estimator.fuse_airspeed(time_now_usec, airspeed_true_raw, vI, Vector2f{vel_var(0), vel_var(1)});
+	}
+}
+
+// this function returns the current states of the wind estimator to be published in the airspeed module
+wind_estimate_s
+AirspeedValidator::get_wind_estimator_states(uint64_t timestamp)
+{
+	wind_estimate_s wind_est = {};
+
+	wind_est.timestamp = timestamp;
+	float wind[2];
+	_wind_estimator.get_wind(wind);
+	wind_est.windspeed_north = wind[0];
+	wind_est.windspeed_east = wind[1];
+	float wind_cov[2];
+	_wind_estimator.get_wind_var(wind_cov);
+	wind_est.variance_north = wind_cov[0];
+	wind_est.variance_east = wind_cov[1];
+	wind_est.tas_innov = _wind_estimator.get_tas_innov();
+	wind_est.tas_innov_var = _wind_estimator.get_tas_innov_var();
+	wind_est.beta_innov = _wind_estimator.get_beta_innov();
+	wind_est.beta_innov_var = _wind_estimator.get_beta_innov_var();
+	wind_est.tas_scale = _wind_estimator.get_tas_scale();
+	return wind_est;
+}
+
+void
+AirspeedValidator::update_EAS_scale()
+{
+	if (_wind_estimator_scale_estimation_on) {
+		_EAS_scale = 1.0f / math::constrain(_wind_estimator.get_tas_scale(), 0.75f, 1.25f);
+
+	} else {
+		_EAS_scale = _airspeed_scale_manual;
+	}
+
+}
+
+void
+AirspeedValidator::update_EAS_TAS(float air_pressure_pa, float air_temperature_celsius)
+{
+	_EAS = calc_EAS_from_IAS(_IAS, _EAS_scale);
+	_TAS = calc_TAS_from_EAS(_EAS, air_pressure_pa, air_temperature_celsius);
+}
+
+void
+AirspeedValidator::check_airspeed_innovation(uint64_t time_now, float estimator_status_vel_test_ratio,
+		float estimator_status_mag_test_ratio)
+{
+	// Check normalised innovation levels with requirement for continuous data and use of hysteresis
+	// to prevent false triggering.
+
+	if (_wind_estimator.get_wind_estimator_reset()) {
+		_time_wind_estimator_initialized = time_now;
+	}
+
+	/* Wait 2s after wind estiamtor was initialzed or reset until innovation checks are done.
+		TO DO: base this on actual wind states variance, not just time.*/
+	if (!_in_fixed_wing_flight || (hrt_elapsed_time(&_time_wind_estimator_initialized) < 2000_ms)) {
+		// not in a flight condition that enables use of this check, thus pass check
+		_innovations_check_failed = false;
+		_time_last_tas_pass = time_now;
+		_time_last_tas_fail = 0;
+		_airspeed_valid = true;
+		_time_last_aspd_innov_check = time_now;
+
+	} else {
+		float dt_s = math::constrain((time_now - _time_last_aspd_innov_check) / 1e6f, 0.01f, 1.0f); // limit to 1Hz...100Hz
```

Comment:
```suggestion
		float dt_s = math::max((time_now - _time_last_aspd_innov_check) / 1e6f, 0.01f); // limit to 100Hz
```

Since you check below for `if (dt_s < 1.0f) {`